nilearn/maskers/nifti_labels_masker.py

```diff
@@ -604,7 +634,24 @@ def transform_single_imgs(self, imgs, confounds=None, sample_mask=None):
             verbose=self.verbose,
         )
 
-        self.labels_ = labels_
+        self.labels_ = ids
+
+        # defining a dictionary containing regions ids
+        region_ids = {'background': self.background_label}
+        for i in range(region_signals.shape[1]):
+            # ids does not include background label
+            region_ids[i] = ids[i]
```

Comment:
It's a bit confusing having "background" key map to a 0 label and then a 0 key map to a 1 label. Why have this and region names dictionary? I guess in the end what's useful to the user is one dictionary mapping masked labels to their names (or arbitrary names matching the label if self.labels is None)

Reply:
The `region_ids_` is indeed needed and it was also mentioned in [the original issue](https://github.com/nilearn/nilearn/issues/3085). The main reason is the empty regions. So in brief, the signal in the ith column does not necessarily correspond to the ith region. We could align `region_names_` and `region_signals` so that the ith signal corresponds to the ith name, but another problem is that the `labels_img` or `region_atlas_` that we are going to return is filled with the original ids which do not correspond to the new order of ids. We could update the ids in `region_atlas_` too, but besides all these @jeromedockes thinks it's a good idea that we have map to the original ids for users reference or some correspondence between the new and original ids, and I agree. What do you think?
But I can remove background key if it is confusing. I just thought it's good to have it.